Sources/ArcGISToolkit/Components/FloatingPanel/FloatingPanel.swift

```diff
@@ -48,19 +48,18 @@ struct FloatingPanel<Content>: View where Content: View {
     /// The height of the content.
     @State private var height: CGFloat = .minHeight
     
-    /// A Boolean value indicating whether the keyboard is open.
-    @State private var keyboardIsOpen = false
+    /// The current height of the device keyboard.
+    @State private var keyboardHeight: CGFloat = 0
+    
+    /// The current state of the device keyboard.
+    @State private var keyboardState: KeyboardState = .closed
     
     /// The latest recorded drag gesture value.
     @State private var latestDragGesture: DragGesture.Value?
     
     /// The maximum allowed height of the content.
     @State private var maximumHeight: CGFloat = .zero
     
-    /// A Boolean value indicating whether the resignFirstResponder should be sent for the current
-    /// drag gesture.
-    @State private var shouldSendResign = true
```

Comment:
No need to track if we've sent the dismiss any longer because it's only done when the gesture ends now and that runs once.